apps/api/src/controllers/applicationController.ts

```diff
@@ -252,3 +256,57 @@ export const submitRevision = async ({ applicationId }: { applicationId: number
 		return failure(message, error);
 	}
 };
+
+export const requestApplicationRevisions = async ({
+	applicationId,
+	role,
+	revisionData,
+}: {
+	applicationId: number;
+	role: string;
+	revisionData: RevisionRequestModel;
+}): AsyncResult<JoinedApplicationRecord> => {
+	try {
+		const database = getDbInstance();
+		const service: ApplicationService = applicationSvc(database);
+
+		const result = await service.getApplicationById({ id: applicationId });
+
+		if (!result.success) {
+			return result;
+		}
+
+		const application = result.data;
+		const appStateManager = new ApplicationStateManager(application);
+
+		if (
+			(role === 'DAC_MEMBER' && application.state !== ApplicationStates.DAC_REVIEW) ||
+			(role === 'INSTITUTIONAL_REP' && application.state !== ApplicationStates.INSTITUTIONAL_REP_REVIEW)
+		) {
+			return failure('Application is not in the correct status for revisions.');
+		}
+
+		let revisionResult;
+
+		if (role === 'DAC_MEMBER') {
+			revisionResult = await appStateManager.reviseDacReview();
+		} else {
+			revisionResult = await appStateManager.reviseRepReview();
+		}
```

Comment:
Discussed with Jon and we should break this into two separate functions because each endpoint is role based

So use only one half of this branching `appStateManager` logic in each function